src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -340,7 +399,111 @@ bool SQLStorage::loadTlsPkey(std::string* pkey) {
+    LOG_ERROR << "Can't clear metadata: " << db.errmsg();
+    return;
+  }
+
+  auto ins_statement = db.prepareStatement<SQLBlob, int, int, int>(
+      "INSERT INTO meta VALUES (?, ?, ?, ?);", SQLBlob(data), static_cast<int>(repo), role.ToInt(), version.version());
+
+  if (ins_statement.step() != SQLITE_DONE) {
+    LOG_ERROR << "Can't add metadata: " << db.errmsg();
+    return;
+  }
+
+  db.commitTransaction();
+}
+
+bool SQLStorage::loadRole(std::string* data, Uptane::RepositoryType repo, Uptane::Role role, Uptane::Version version) {
+  SQLite3Guard db(config_.sqldb_path.c_str());
+  if (db.get_rc() != SQLITE_OK) {
+    LOG_ERROR << "Can't open database: " << db.errmsg();
+    return false;
+  }
+
+  request = SQLReqId::kGetTable;
+  req_response_table.clear();
+
+  // version < 0 => latest metadata requested
+  if (version.version() < 0) {
+    auto statement = db.prepareStatement<int, int>(
+        "SELECT meta FROM meta WHERE (repo=? AND meta_type=?) ORDER BY version DESC LIMIT 1;", static_cast<int>(repo),
+        role.ToInt());
+    int result = statement.step();
+
+    if (result == SQLITE_DONE) {
+      LOG_TRACE << "Meta not present";
+      return false;
+    } else if (result != SQLITE_ROW) {
+      LOG_ERROR << "Can't get meta: " << db.errmsg();
+      return false;
+    }
+    *data = std::string(reinterpret_cast<const char*>(sqlite3_column_blob(statement.get(), 0)));
+  } else {
+    auto statement =
+        db.prepareStatement<int, int, int>("SELECT meta FROM meta WHERE (repo=? AND meta_type=? AND version=?);",
+                                           static_cast<int>(repo), role.ToInt(), version.version());
+
+    int result = statement.step();
+
+    if (result == SQLITE_DONE) {
+      LOG_TRACE << "Meta not present";
+      return false;
+    } else if (result != SQLITE_ROW) {
+      LOG_ERROR << "Can't get meta: " << db.errmsg();
+      return false;
+    }
+    *data = std::string(reinterpret_cast<const char*>(sqlite3_column_blob(statement.get(), 0)));
+  }
+
+  return true;
+}
+
+void SQLStorage::clearNonRootMeta(Uptane::RepositoryType repo) {
+  SQLite3Guard db(config_.sqldb_path.c_str());
+  if (db.get_rc() != SQLITE_OK) {
+    LOG_ERROR << "Can't open database: " << db.errmsg();
+    return;
+  }
+
+  auto del_statement =
+      db.prepareStatement<int>("DELETE FROM meta WHERE (repo=? AND meta_type != 0);", static_cast<int>(repo));
+
+  if (del_statement.step() != SQLITE_DONE) {
+    LOG_ERROR << "Can't clear metadata: " << db.errmsg();
+  }
+}
+
+void SQLStorage::clearMetadata() {
+  SQLite3Guard db(config_.sqldb_path.c_str());
+  if (db.get_rc() != SQLITE_OK) {
+    LOG_ERROR << "Can't open database: " << db.errmsg();
+    return;
+  }
+
+  if (db.exec("DELETE FROM meta;", nullptr, nullptr) != SQLITE_OK) {
+    LOG_ERROR << "Can't clear metadata: " << db.errmsg();
+    return;
+  }
+}
+/*void SQLStorage::storeMetadataCommon(const Uptane::RawMetaPack& metadata, const std::string& tablename) {
```

Comment:
Should we remove all this commented code?

Reply:
Yes, sure, I could have missed something.